src/metabase/lib/native.cljc

```diff
@@ -15,25 +15,96 @@
    [metabase.util.malli :as mu]
    [metabase.util.malli.registry :as mr]))
 
+(def tokens
+  "The tokens used for template tags."
+  ["{{" "}}" "[[" "]]"])
+
+(defn- next-token [text start]
+  (let [indexes (map #(or (str/index-of text % start) ##Inf) tokens)
+        index   (apply min indexes)]
+    (if (infinite? index)
+      [nil nil]
+      [index (subs text index (+ 2 index))])))
+
+(mu/defn ^:private tokenize-query :- [:sequential :string]
+  "Tokenize the query for easier parsing.
+   This splits the string at the following tokens: {{ }} [[ ]],
+   and keeps the tokens in the result."
```

Comment:
I suggest calling tokens all relevant parts of the input and separators the double brackets. I also suggest referring to the var `tokens` instead of repeating them in the docstring.